### PR TITLE
stellar-etl: implement export_accounts command

### PR DESCRIPTION
**Pull request by Isaiah-Turner**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).
</details>

### What
This PR implements some of the export_accounts command. It allows users to connect to the history archives, retrieve account information starting from the genesis ledger, and exports the JSON encodings to an output file. 

### Why
This command is part of the larger CLI tool. It will allow for updating the BigQuery table history_accounts more efficiently.

### Known limitations
Right now, the bucket list is used to get all the information starting from the genesis ledger, but it's difficult to get arbitrary ranges. In the next PR, the command will use captive core in order to update information in arbitrary ranges. 

The start-time and end-time flags are not implemented. The ingestion system uses sequence numbers instead of times. If it is expanded to use times, then we can add the flags. The serialize-method flag also is not implemented, since at the moment we only have json encodings for the structs.

### CODE REVIEW

File: cmd/export_accounts.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/stellar/stellar-etl/internal/transform"
+	"github.com/stellar/stellar-etl/internal/utils"
+)
+
+var accountsCmd = &cobra.Command{
+	Use:   "export_accounts",
+	Short: "Exports the account data.",
+	Long:  `Exports historical account data within the specified range to an output file.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		startNum, endNum, limit, path, useStdOut := utils.MustBasicFlags(cmd.Flags(), cmdLogger)
```
Comment (debnil) [inline cmd/export_accounts.go:19]: Naming convention: `stdout` is a phrase, so the variable should be named `useStdout`.

File: cmd/export_accounts.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+				fmt.Println(string(marshalled))
+			}
+		}
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(accountsCmd)
+	utils.AddBasicFlags("accounts", accountsCmd.Flags())
+	ledgersCmd.MarkFlagRequired("end-ledger")
```
Comment (debnil) [inline cmd/export_accounts.go:55]: Not sure I follow this. Why are you modifying `ledgersCmd`? Or should this be `accountsCmd`?

File: internal/input/accounts.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	"github.com/stellar/go/exp/ingest/adapters"
+	ingestio "github.com/stellar/go/exp/ingest/io"
+	"github.com/stellar/go/historyarchive"
+	"github.com/stellar/go/xdr"
+)
+
+// GetAccounts returns a slice of ledger close metas for the ledgers in the provided range (inclusive on both ends)
+func GetAccounts(start, end uint32, limit int64) ([]xdr.LedgerEntry, error) {
+	archiveStellarURL := "http://history.stellar.org/prd/core-live/core_live_001"
```
Comment (debnil) [inline internal/input/accounts.go:16]: Pull this out of the function as a package-wide variable - it'll be helpful for other commands.

File: internal/input/accounts.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+		return readBucketList(archive, checkpointSeq, xdr.LedgerEntryTypeAccount)
+	}
+
+	// Use captive core to read a range otherwise
+	// TODO: connect to captive core and read from there
+	return readFromCaptive(start, end)
+}
+
+func readFromCaptive(start, end uint32) ([]xdr.LedgerEntry, error) {
```
Comment (debnil) [inline internal/input/accounts.go:51]: Nit: Add a TODO in the function body, indicating it needs to be implemented.

File: internal/input/accounts.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+	return []xdr.LedgerEntry{}, nil
+}
+
+func readBucketList(archive *historyarchive.Archive, checkpointSeq uint32, entryType xdr.LedgerEntryType) ([]xdr.LedgerEntry, error) {
```
Comment (debnil) [inline internal/input/accounts.go:55]: Function looks great! Awesome job thinking through effective use of the bucket list. Very readable and understandable :)

File: cmd/export_accounts.go (hunk 2 above)
Comment (Isaiah-Turner) [reply cmd/export_accounts.go:55]: Good point. That should be accountsCmd.

**Pull request merged**